src/plugins/kibana/public/settings/sections/indices/add_data_steps/pattern_review_step.js

```diff
@@ -45,12 +48,13 @@ modules.get('apps/settings')
           id: 'filebeat-*',
           title: 'filebeat-*',
           timeFieldName: pickDefaultTimeFieldName(this.dateFields),
-          fields: _.map(this.sampleDoc, (value, key) => {
-            let type = knownFieldTypes[key] || typeof value;
-            if (type === 'object' && _.isArray(value) && !_.isEmpty(value)) {
-              type = typeof value[0];
+          fields: _.map(fields, (fieldName) => {
+            const fieldValue = _.get(this.sampleDoc, fieldName);
+            let type = knownFieldTypes[fieldName] || typeof fieldValue;
+            if (type === 'object' && _.isArray(fieldValue) && !_.isEmpty(fieldValue)) {
```

Comment:
Is there a reason for testing for `object` here? Isn't `_.isArray(fieldValue) && !_.isEmpty(fieldValue)` sufficient?